pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -6805,3 +6805,232 @@ func TestDashbuilderWithPartialCORS(t *testing.T) {
+	envs := env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	controllerStrategyEnabled := false
+	for _, env := range envs {
+
+		if strings.HasPrefix(env.Name, "KIE_SERVER_CONTROLLER_OPENSHIFT_ENABLED") {
+			assert.Equal(t, env.Value, "false")
+			controllerStrategyEnabled = true
+		}
+	}
+	assert.True(t, controllerStrategyEnabled)
+}
+
+func checkConsoleOpenshiftStrategyAssertions(t *testing.T, cr *api.KieApp, assertion bool) {
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	envs := env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+
+	openshiftStrategyEnabled := false
+	openshiftStrategyDiscoveryEnabled := false
+	openshiftStrategyPreferEnabled := false
+	openshiftStrategyCacheTTL := false
+	openshiftStrategyEnabled, openshiftStrategyDiscoveryEnabled, openshiftStrategyPreferEnabled, openshiftStrategyCacheTTL = checkConsoleEnvsWithOpenshiftStartupStrategy(t, envs, openshiftStrategyEnabled, openshiftStrategyDiscoveryEnabled, openshiftStrategyPreferEnabled, openshiftStrategyCacheTTL)
+	if assertion {
+		assert.True(t, openshiftStrategyEnabled)
+		assert.True(t, openshiftStrategyDiscoveryEnabled)
+		assert.True(t, openshiftStrategyPreferEnabled)
+		assert.True(t, openshiftStrategyCacheTTL)
+	} else {
+		assert.False(t, openshiftStrategyEnabled)
+		assert.False(t, openshiftStrategyDiscoveryEnabled)
+		assert.False(t, openshiftStrategyPreferEnabled)
+		assert.False(t, openshiftStrategyCacheTTL)
+	}
+}
+
+func checkConsoleEnvsWithOpenshiftStartupStrategy(t *testing.T, envs []corev1.EnvVar, openshiftStrategyEnabled bool, openshiftStrategyDiscoveryEnabled bool, openshiftStrategyPreferEnabled bool, openshiftStrategyCacheTTL bool) (bool, bool, bool, bool) {
+	for _, env := range envs {
+
+		if strings.HasPrefix(env.Name, "KIE_SERVER_CONTROLLER_OPENSHIFT_ENABLED") {
+			assert.Equal(t, env.Value, "true")
+			openshiftStrategyEnabled = true
+		}
+
+		if strings.HasPrefix(env.Name, "KIE_SERVER_CONTROLLER_OPENSHIFT_GLOBAL_DISCOVERY_ENABLED") {
+			assert.Equal(t, env.Value, "true")
+			openshiftStrategyDiscoveryEnabled = true
+		}
+
+		if strings.HasPrefix(env.Name, "KIE_SERVER_CONTROLLER_OPENSHIFT_PREFER_KIESERVER_SERVICE") {
+			assert.Equal(t, env.Value, "true")
+			openshiftStrategyPreferEnabled = true
+		}
+
+		if strings.HasPrefix(env.Name, "KIE_SERVER_CONTROLLER_TEMPLATE_CACHE_TTL") {
+			assert.Equal(t, env.Value, "5000")
+			openshiftStrategyCacheTTL = true
+		}
+	}
+	return openshiftStrategyEnabled, openshiftStrategyDiscoveryEnabled, openshiftStrategyPreferEnabled, openshiftStrategyCacheTTL
+}
+
+func TestConsoleOpenshitStartupStrategyConfiguration(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamAuthoring,
+			CommonConfig: api.CommonConfig{
+				StartupStrategy: createStartupStrategy(api.OpenshiftStartupStrategy, 5000),
```

Comment:
maybe set a different value than the default?

Reply:
ok